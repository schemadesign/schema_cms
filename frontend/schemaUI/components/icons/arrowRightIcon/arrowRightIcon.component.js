import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import ArrowRightSVG from '../../../images/icons/arrow-right.svg';
import { containerStyles } from './arrowRightIcon.styles';

export class ArrowRightIcon extends PureComponent {
  static propTypes = {
    customStyles: PropTypes.object,
  };

  render() {
    const { customStyles, ...restProps } = this.props;
    const styles = { ...containerStyles, ...customStyles };

    return <ArrowRightSVG {...restProps} style={styles} />;
  }
}
