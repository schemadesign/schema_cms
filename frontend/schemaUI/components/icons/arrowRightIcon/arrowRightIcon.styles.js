import { light } from '../../../utils/theme';

export const containerStyles = {
  fill: light.iconDark,
  width: '60px',
  height: '60px',
};
