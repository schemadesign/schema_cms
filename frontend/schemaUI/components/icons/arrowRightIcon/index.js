export { ArrowRightIcon } from './arrowRightIcon.component';
