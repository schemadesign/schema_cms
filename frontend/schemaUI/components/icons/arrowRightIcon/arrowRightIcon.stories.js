import React from 'react';
import { storiesOf } from '@storybook/react';

import { ArrowRightIcon } from './arrowRightIcon.component';
import { withTheme } from '../../../.storybook/decorators';

storiesOf('Icons/ArrowRightIcon', module).add('Default', () => <ArrowRightIcon />);

const defaultProps = { customStyles: { fill: 'blue' } };
storiesOf('Icons/ArrowRightIcon', module)
  .addDecorator(withTheme())
  .add('with custom styles', () => <ArrowRightIcon {...defaultProps} />);
