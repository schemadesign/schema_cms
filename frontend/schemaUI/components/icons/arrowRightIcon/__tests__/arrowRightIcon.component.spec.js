import React from 'react';
import { shallow } from 'enzyme';

import { ArrowRightIcon } from '../arrowRightIcon.component';

describe('ArrowRightIcon: Component', () => {
  const defaultProps = {};

  const component = props => <ArrowRightIcon {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render with custom styles', () => {
    const customStyles = {
      fill: 'blue',
    };

    const wrapper = render({ customStyles });
    global.expect(wrapper).toMatchSnapshot();
  });
});
