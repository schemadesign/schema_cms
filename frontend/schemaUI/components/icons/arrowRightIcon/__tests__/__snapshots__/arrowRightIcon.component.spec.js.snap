// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowRightIcon: Component should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`ArrowRightIcon: Component should render with custom styles 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
